e2e/cephfs.go

```diff
@@ -172,549 +194,541 @@ var _ = Describe("cephfs", func() {
-					validateNormalUserPVCAccess(pvcPath, f)
-				})
+			By("create/delete multiple PVCs and Apps", func() {
+				totalCount := 2
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load pvc %v", err)
+				}
+				pvc.Namespace = f.UniqueName
 
-				By("create/delete multiple PVCs and Apps", func() {
-					totalCount := 2
-					pvc, err := loadPVC(pvcPath)
+				app, err := loadApp(appPath)
+				if err != nil {
+					e2elog.Failf("failed to load application %v", err)
+				}
+				app.Namespace = f.UniqueName
+				// create pvc and app
+				for i := 0; i < totalCount; i++ {
+					name := fmt.Sprintf("%s%d", f.UniqueName, i)
+					err = createPVCAndApp(name, f, pvc, app, deployTimeout)
 					if err != nil {
-						Fail(err.Error())
+						e2elog.Failf("failed to create pvc or application %v", err)
 					}
-					pvc.Namespace = f.UniqueName
 
-					app, err := loadApp(appPath)
+				}
+
+				validateSubvolumeCount(f, totalCount, fileSystemName, subvolumegroup)
+				// delete pvc and app
+				for i := 0; i < totalCount; i++ {
+					name := fmt.Sprintf("%s%d", f.UniqueName, i)
+					err = deletePVCAndApp(name, f, pvc, app)
 					if err != nil {
-						Fail(err.Error())
+						e2elog.Failf("failed to delete pvc or application %v", err)
 					}
-					app.Namespace = f.UniqueName
-					// create pvc and app
-					for i := 0; i < totalCount; i++ {
-						name := fmt.Sprintf("%s%d", f.UniqueName, i)
-						err := createPVCAndApp(name, f, pvc, app, deployTimeout)
-						if err != nil {
-							Fail(err.Error())
-						}
 
-					}
-					subVol := listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
-					if len(subVol) != totalCount {
-						msg := fmt.Sprintf("subvolumes %v. subvolume count %d not matching expected count %v", subVol, len(subVol), totalCount)
-						e2elog.Logf(msg)
-						Fail(msg)
-					}
-					// delete pvc and app
-					for i := 0; i < totalCount; i++ {
-						name := fmt.Sprintf("%s%d", f.UniqueName, i)
-						err := deletePVCAndApp(name, f, pvc, app)
-						if err != nil {
-							Fail(err.Error())
-						}
+				}
+				validateSubvolumeCount(f, 0, fileSystemName, subvolumegroup)
+			})
 
-					}
-					subVol = listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
-					if len(subVol) != 0 {
-						msg := fmt.Sprintf("subvolumes %v. subvolume count %d not matching expected count %v", subVol, len(subVol), 0)
-						e2elog.Logf(msg)
-						Fail(msg)
-					}
-				})
+			By("check data persist after recreating pod with same pvc", func() {
+				err := checkDataPersist(pvcPath, appPath, f)
+				if err != nil {
+					e2elog.Failf("failed to check data persist in pvc %v", err)
+				}
+			})
 
-				By("check data persist after recreating pod with same pvc", func() {
-					err := checkDataPersist(pvcPath, appPath, f)
-					if err != nil {
-						Fail(err.Error())
-					}
-				})
+			By("creating a PVC, deleting backing subvolume, and checking successful PV deletion", func() {
```

Comment:
create PVC, delete backing subvolume and check pv deletion sound better , Isnt it?